scripts/apps/master-desk/components/FilterPanelComponent.tsx

```diff
@@ -34,6 +36,26 @@ export class FilterPanelComponent extends React.Component<IProps, IState> {
         this.handleInputChange = this.handleInputChange.bind(this);
         this.addFilter = this.addFilter.bind(this);
         this.clearDeskFilter = this.clearDeskFilter.bind(this);
+        this.clearFilters = this.clearFilters.bind(this);
+    }
+
+    componentDidUpdate(prevProps: IProps) {
```

Comment:
When state is tightly connected to props and needs to be updated when props change, it's cleaner to use a key instead. You wouldn't need `componentDidUpdate` and the component would re-initialize when filters change.

![2021-06-04_09-51](https://user-images.githubusercontent.com/2076953/120767242-4eb1dc00-c51b-11eb-8276-c9a8f228c9cf.png)

Reply:
Yep, it does initialize the state again, but since your state is derived from props, that shouldn't be problem if you add a few lines in the constructor to initialize state from props. I'm not 100% sure, but looks like it should work.